flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/data/conversion/RowRowConverter.java

```diff
@@ -58,22 +67,48 @@ public void open(ClassLoader classLoader) {
 	public RowData toInternal(Row external) {
 		final int length = fieldConverters.length;
 		final GenericRowData genericRow = new GenericRowData(external.getKind(), length);
-		for (int pos = 0; pos < length; pos++) {
-			final Object value = external.getField(pos);
-			genericRow.setField(pos, fieldConverters[pos].toInternalOrNull(value));
+
+		final Set<String> fieldNames = external.getFieldNames(false);
```

Comment:
It would be interesting to see in benchmarks what the impact of this is, and of name-based field access in general.